src/tc/actions/message.rs

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: MIT
+
+use anyhow::Context;
+use netlink_packet_utils::nla::{DefaultNla, NlaBuffer};
+use netlink_packet_utils::nla::{Nla, NlasIterator};
+use netlink_packet_utils::{DecodeError, Emitable, Parseable};
+
+use crate::tc::actions::{TcActionMessageBuffer, TcActionMessageHeader};
+use crate::tc::TcAction;
+
+/// Message to describe [tc-actions]
+///
+/// [tc-actions]: https://man7.org/linux/man-pages/man8/tc-actions.8.html
+#[derive(Debug, PartialEq, Eq, Clone, Default)]
+#[non_exhaustive]
+pub struct TcActionMessage {
+    /// Header of the message.
+    pub header: TcActionMessageHeader,
+    /// Attributes of the message.
+    pub attributes: Vec<TcActionMessageAttribute>,
+}
+
+const TCA_ACT_FLAG_LARGE_DUMP_ON: u32 = 1 << 0;
+const TCA_ACT_FLAG_TERSE_DUMP: u32 = 1 << 1;
+
+bitflags! {
+    /// Flags to configure action dumps (list operations).
+    #[derive(Debug, PartialEq, Eq, Clone, Copy, Default, PartialOrd, Ord, Hash)]
+    #[non_exhaustive]
+    pub struct TcActionMessageFlags: u32 {
+        /// From `iproute2`'s [`rtnetlink.h`]
+        ///
+        /// If set, this flag enables more than TCA_ACT_MAX_PRIO actions in a single
+        /// actions listing operation.
+        ///
+        /// > TCA_ACT_FLAG_LARGE_DUMP_ON user->kernel to request for larger than
+        ///   TCA_ACT_MAX_PRIO actions in a dump.
+        ///   All dump responses will contain the number of actions being dumped
+        ///   stored in for user app's consumption in TCA_ROOT_COUNT
+        ///
+        /// [`rtnetlink.h`]: https://github.com/iproute2/iproute2/blob/89210b9ec1c445ae963d181b5816d12a0cdafbb6/include/uapi/linux/rtnetlink.h#L803-L806
```

Comment:
Those link might not accessible in the future. No need to mention it.

iproute2 might be the de-factor netlink client, but I prefer explains stuff using kernel codes or function name.

Reply:
Ok.  Will fix.